python/plugins/processing/algs/gdal/rasterize.py

```diff
@@ -170,6 +172,9 @@ def commandName(self):
 
     def getConsoleCommands(self, parameters, context, feedback, executing=True):
         ogrLayer, layerName = self.getOgrCompatibleSource(self.INPUT, parameters, context, feedback, executing)
+        source = self.parameterAsSource(parameters, self.INPUT, context)
+        if source is None:
```

Comment:
This check should be made before any attempts to access input, as we want to fail as early as possible.

Reply:
That check is there also for other GDAL algs: [Buffer vectors](https://github.com/qgis/QGIS/blob/5339d62715f4faaf0ed04bdec97335a7abecdd7c/python/plugins/processing/algs/gdal/Buffer.py#L106-L109), [Clip vector by extent](https://github.com/qgis/QGIS/blob/5339d62715f4faaf0ed04bdec97335a7abecdd7c/python/plugins/processing/algs/gdal/ClipVectorByExtent.py#L81-L85), [Dissolve](https://github.com/qgis/QGIS/blob/5339d62715f4faaf0ed04bdec97335a7abecdd7c/python/plugins/processing/algs/gdal/Dissolve.py#L116-L119), and so on... See https://github.com/qgis/QGIS/commit/5339d62715f4faaf0ed04bdec97335a7abecdd7c

Reply:
I know.

And I am not arguing about whether we should have it or not. I am only saying that you should check parameter for validity **before** accessing this parameter.

Reply:
Thanks for clarification. Done. So the same change should be made to the "Clip vector by extent" alg.
https://github.com/qgis/QGIS/blob/8563a5eb43e0461a24bab302b713defbf980fde3/python/plugins/processing/algs/gdal/ClipVectorByExtent.py#L76-L80